game/screen_paths.cc

```diff
@@ -8,66 +8,151 @@
 #include <boost/thread.hpp>
 #include <boost/bind.hpp>
 
-ScreenPaths::ScreenPaths(std::string const& name, Audio& audio): Screen(name), m_audio(audio) {}
+ScreenPaths::ScreenPaths(std::string const& name, Audio& audio, Songs& songs): Screen(name), m_audio(audio), m_songs(songs) {}
 
 void ScreenPaths::enter() {
 	m_theme.reset(new ThemeAudioDevices());
-	m_txtinp.text.clear();
-
-	// FIXME: Temp error message
-	Game::getSingletonPtr()->dialog(
-		_("This tool is not yet available.\n"
-		  "Configure paths by adding them\n"
-		  "as command line parameters and\n"
-		  "then save them in configuration menu."));
+	fs::path homedir(getenv("HOME"));
+	generateMenuFromPath(homedir);
 }
 
-void ScreenPaths::exit() { m_theme.reset(); }
+void ScreenPaths::exit() {
+	m_theme.reset();
+	m_songs.reload();
+}
 
 void ScreenPaths::manageEvent(SDL_Event event) {
-	if (event.type == SDL_TEXTINPUT) {
-		m_txtinp += event.text.text;
-	} else if (event.type == SDL_KEYDOWN) {
+	if (event.type == SDL_KEYDOWN) {
 		SDL_Keycode key = event.key.keysym.scancode;
 		uint16_t modifier = event.key.keysym.mod;
 		// Reset to defaults
 		if (key == SDL_SCANCODE_R && modifier & KMOD_CTRL) {
 			config["paths/songs"].reset(modifier & KMOD_ALT);
 			config["paths/system"].reset(modifier & KMOD_ALT);
 			// TODO: Save
+		}else if (key == SDL_SCANCODE_S && modifier & KMOD_CTRL) {
+			writeConfig(modifier & KMOD_ALT);
+			Game::getSingletonPtr()->flashMessage((modifier & KMOD_ALT)
+				? _("Settings saved as system defaults.") : _("Settings saved."));
 		}
 	}
 }
 
 void ScreenPaths::manageEvent(input::NavEvent const& ev) {
 	Game* gm = Game::getSingletonPtr();
 	if (ev.button == input::NAV_CANCEL) {
-		if (m_txtinp.text.empty()) gm->activateScreen("Intro");
-		else m_txtinp.text.clear();
+		gm->activateScreen("Intro");
 	}
 	else if (ev.button == input::NAV_PAUSE) m_audio.togglePause();
-	else if (ev.button == input::NAV_START) { 
-		// TODO: Save config
-		gm->activateScreen("Intro");
+	else if (ev.button == input::NAV_DOWN) m_menu.move(1);
```

Comment:
Magic numbers.. It's the best to avoid these and define them as consts at the top of the class. Same goes for line 48, 49, 50